libs/mep/mwpm-143159/timeline/timeline.js

```diff
@@ -0,0 +1,153 @@
+import { createTag } from '../../../utils/utils.js';
+
+function isColor(str) {
+  const hexRegex = /^#(?:[0-9a-fA-F]{3}){1,2}$/;
+  const rgbRegex = /^rgb\(\s*\d{1,3}\s*,\s*\d{1,3}\s*,\s*\d{1,3}\s*\)$/;
+  return hexRegex.test(str) || rgbRegex.test(str);
+}
+
+function isGradient(str) {
+  return str.startsWith('linear-gradient');
+}
+
+function isColorOrGradient(str) {
+  return isColor(str) || isGradient(str);
+}
+
+function getColWidth(text, colWidths) {
+  const numRegex = /\b\d{1,3}\b/;
+  colWidths.push((text.match(numRegex) || [])[0]);
+}
+function createRow() {
+  return [createTag('div', { class: 'row' }),
+    createTag('div', { class: 'left' }),
+    createTag('div', { class: 'right' }),
+  ];
+}
+function createBars(index) {
+  return index === 0
+    ? [createTag('div', { class: 'bar' })] : [createTag('div', { class: 'bar' }), createTag('div', { class: 'bar' })];
+}
+function addBarRow() {
+  const [barRow, left, right] = createRow();
+  const sides = [left, right];
+  sides.forEach((text, index) => {
+    sides[index].append(...createBars(index));
+    barRow.append(sides[index]);
+  });
+  return barRow;
+}
+function addBottomRow(periodText) {
+  const [periodRow, left, right] = createRow();
+  const sides = [left, right];
+  periodText.forEach((text, index) => {
+    sides[index].append(createTag('p', { class: 'period body-s' }, text));
+    periodRow.append(sides[index]);
+  });
+  return periodRow;
+}
+
+function setBG(el, color) {
+  el.style.background = color;
+}
+
+function updateForRTL(lgStr, el) {
+  if (el.closest('[dir="rtl"]')) {
+    return lgStr.replace(/to right|to left|90deg|180deg|turn 0.25|turn 0.75/g, (match) => {
+      const converter = {
```

Comment:
wondering if this could be achieved with CSS instead

Reply:
I'll have to think of this a bit more but so far, I can't think of a css-only way to approach this.